### ISSUE TITLE
Characters at the bottom of the table are not properly replaced when there is a link in column

### ISSUE THREAD

**Issue by baratali** [Fuco1/org-pretty-table#12]
Thank you for this awesome package!
I noticed that some characters at the bottom of the table are not properly replaced when there is a link in column.
They're replaced correctly if I remove the link.
![image](https://github.com/Fuco1/org-pretty-table/assets/3431491/63b6805d-a8af-4de6-b988-89d24249de2b)

Here is plain text:
```
|---------+------+---------+---------+---------|
| column1 | [[https://github.com/][link]] | column2 | column3 | column4 |
|---------+------+---------+---------+---------|

|---------+------+---------+---------+---------|
| column1 | link | column2 | column3 | column4 |
|---------+------+---------+---------+---------|
```

**Comment by Fuco1** [comment]
Hmm, yea.  The way it works now is it looks above the `+` to see if there's a `|`.  We can add a special case for the end of table but I guess it would then break in the middle.  I'm not sure if it's always safe to replace `+` with the cross.

**Comment by ermingol23** [comment]
Replacing all '+' with a cross could get interesting if you have a table cell containing a mathematical expression, for instance
```
|---------+------+---------+---------+---------|
| column1 | 10+2 | column2 | column3 | column4 |
|---------+------+---------+---------+---------|
```
Some more advanced algorithm need to be implemented to understand whether a "table token" is indeed part of the table or not, for instance if you encounter a '+' a check could be made if the entire line consist of "table tokens". The '|' characters are more complicated as you could have a table like this
```
|---------+-----+---------+---------+---------|
| column1 | A|B | column2 | column3 | column4 |
|---------+-----+---------+---------+---------|
```
where the '|' between 'A' and 'B' should not be replaced. Hence the algorithm need to understand that there is a link and then only consider the 'link' part of '[[https://github.com/][link]]' when moving up/down one line to find a matching '+'. This could get more complicated fast if you also want to handle cases like this
```
|---------+-----------+---------+---------+---------|
| column1 | [[https://github.com/][link]]|[[https://github.com/][link]] | column2 | column3 | column4 |
|---------+-----------+---------+---------+---------|
```
that should render something like
```
|---------+-----------+---------+---------+---------|
| column1 | link|link | column2 | column3 | column4 |
|---------+-----------+---------+---------+---------|
```
Basically you need a Markdown-parser that first strips all the Markdown-specific notation from the table ("renders it") and after that look at the result before determining whether a "table token" belong to the table borders or not. This could be done in a temp-buffer that is then used as a "template" for which characters to change the appearance of. This mode could then be made to dynamically show whether the formatting of the table is correct or not (you would see where to add/remove '-' or space characters to get a table to render properly).

**Comment by Fuco1** [comment]
@ermingol23 This is not supposed to be used with markdown. 

Same situation would happen with any kind of characters which are hidden by org, for example emphasis markers if those are set to invisible etc.  I still don't have a great idea on how to fix it.  Maybe some AI could write some ugly code to do it (it's already ugly tbh).
